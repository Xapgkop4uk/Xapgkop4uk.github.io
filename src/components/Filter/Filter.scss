@import './FilterCurrency/FilterCurrency.scss';
@import './FilterTransfer/FilterTransfer.scss';

.filter-wrapper {
  width: 232px;
  height: 331px;
  background: #ffffff;
  color: #ffffff;
  text-align: left;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  letter-spacing: 0px;
  box-shadow: 0px 1px 4px 0px rgba(91, 137, 164, 0.25);
}