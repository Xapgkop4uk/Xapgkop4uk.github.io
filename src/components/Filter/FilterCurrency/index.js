import FilterCurrency from './FilterCurrency'

export default FilterCurrency
