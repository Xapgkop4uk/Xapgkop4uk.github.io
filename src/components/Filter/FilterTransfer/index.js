import FilterTransfer from './FilterTransfer'

export default FilterTransfer
