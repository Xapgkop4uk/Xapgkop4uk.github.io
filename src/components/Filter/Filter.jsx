import React from 'react'
import propTypes from 'prop-types'
import FilterCurrency from './FilterCurrency'
import FilterTransfer from './FilterTransfer'

const Filter = ({
  handleCurrencyChange,
  selectedCurrency,
  selectedFilters,
  handleFilterChange,
  handleSelectOne
}) => (
  <div className="filter-wrapper">
    <FilterCurrency
      currency={selectedCurrency}
      onClick={handleCurrencyChange}
    />
    <FilterTransfer
      transfersSelected={selectedFilters}
      onClick={handleFilterChange}
      handleSelectOne={handleSelectOne}
    />
  </div>
)

Filter.propTypes = {
  handleCurrencyChange: propTypes.func.isRequired,
  selectedCurrency:     propTypes.string.isRequired,
  selectedFilters:      propTypes.array.isRequired,
  handleFilterChange:   propTypes.func.isRequired,
  handleSelectOne:      propTypes.func.isRequired
}

export default Filter
