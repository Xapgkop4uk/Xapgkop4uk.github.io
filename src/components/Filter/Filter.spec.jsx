import React from 'react'
import renderer from 'react-test-renderer'
import Filter from './Filter'

describe('Filter component', () => {
  let el

  beforeAll(() => {
    el = renderer.create(<Filter
      handleCurrencyChange={() => {}}
      selectedCurrency="RUB"
      selectedFilters={[-1]}
      handleFilterChange={() => {}}
      handleSelectOne={() => {}}
    />)
  })
  it('contains tickets', () => {
    const tree = el.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
