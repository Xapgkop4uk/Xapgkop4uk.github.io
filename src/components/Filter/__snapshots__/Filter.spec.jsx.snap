// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component contains tickets 1`] = `
<div
  className="filter-wrapper"
>
  <div
    className="filter-currency-wrapper"
  >
    <span
      className="currency-label"
    >
      Валюта
    </span>
    <div
      className="currens-selectors"
    >
      <button
        className="selected"
        onClick={[Function]}
        type="button"
      >
        <span
          className="currency"
        >
          RUB
        </span>
      </button>
      <button
        className="general"
        onClick={[Function]}
        type="button"
      >
        <span
          className="currency"
        >
          USD
        </span>
      </button>
      <button
        className="general"
        onClick={[Function]}
        type="button"
      >
        <span
          className="currency"
        >
          EUR
        </span>
      </button>
    </div>
  </div>
  <div
    className="filter-transfer-wrapper"
  >
    <span
      className="transfer-label"
    >
      Количество пересадок
    </span>
    <div
      className="transfer-selectors"
    >
      <div
        className="transfer-checkbox"
      >
        <label
          className="MuiFormControlLabel-root-4"
        >
          <span
            className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiPrivateSwitchBase-root-14 MuiCheckbox-root-8 FilterTransfer-root-1 MuiCheckbox-colorSecondary-13 MuiPrivateSwitchBase-checked-15 MuiCheckbox-checked-9 FilterTransfer-checked-2"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-23"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-27"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <input
                checked={true}
                className="MuiPrivateSwitchBase-input-17"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-71"
            />
          </span>
          <span
            className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-7"
          >
            <span
              className="input-label"
            >
              Все
            </span>
          </span>
        </label>
      </div>
      <div
        className="transfer-checkbox"
      >
        <label
          className="MuiFormControlLabel-root-4"
        >
          <span
            className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiPrivateSwitchBase-root-14 MuiCheckbox-root-8 FilterTransfer-root-1 MuiCheckbox-colorSecondary-13"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-23"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-27"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-17"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-71"
            />
          </span>
          <span
            className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-7"
          >
            <span
              className="input-label"
            >
              Без пересадок
            </span>
          </span>
        </label>
        <button
          className="only-that-button"
          onClick={[Function]}
          type="button"
        >
          ТОЛЬКО
        </button>
      </div>
      <div
        className="transfer-checkbox"
      >
        <label
          className="MuiFormControlLabel-root-4"
        >
          <span
            className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiPrivateSwitchBase-root-14 MuiCheckbox-root-8 FilterTransfer-root-1 MuiCheckbox-colorSecondary-13"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-23"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-27"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-17"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-71"
            />
          </span>
          <span
            className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-7"
          >
            <span
              className="input-label"
            >
              1 пересадка
            </span>
          </span>
        </label>
        <button
          className="only-that-button"
          onClick={[Function]}
          type="button"
        >
          ТОЛЬКО
        </button>
      </div>
      <div
        className="transfer-checkbox"
      >
        <label
          className="MuiFormControlLabel-root-4"
        >
          <span
            className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiPrivateSwitchBase-root-14 MuiCheckbox-root-8 FilterTransfer-root-1 MuiCheckbox-colorSecondary-13"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-23"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-27"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-17"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-71"
            />
          </span>
          <span
            className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-7"
          >
            <span
              className="input-label"
            >
              2 пересадки
            </span>
          </span>
        </label>
        <button
          className="only-that-button"
          onClick={[Function]}
          type="button"
        >
          ТОЛЬКО
        </button>
      </div>
      <div
        className="transfer-checkbox"
      >
        <label
          className="MuiFormControlLabel-root-4"
        >
          <span
            className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiPrivateSwitchBase-root-14 MuiCheckbox-root-8 FilterTransfer-root-1 MuiCheckbox-colorSecondary-13"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-23"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-27"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-17"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-71"
            />
          </span>
          <span
            className="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-7"
          >
            <span
              className="input-label"
            >
              3 пересадки
            </span>
          </span>
        </label>
        <button
          className="only-that-button"
          onClick={[Function]}
          type="button"
        >
          ТОЛЬКО
        </button>
      </div>
    </div>
  </div>
</div>
`;
